cannon/mipsevm/versions/detect_test.go

```diff
@@ -47,28 +46,38 @@ func TestDetectVersion(t *testing.T) {
 			})
 		}
 	}
+}
 
-	// Additionally, check that the latest supported versions write new states in a way that is detected correctly
-	t.Run("SingleThreadedBinary", func(t *testing.T) {
-		state, err := NewFromState(singlethreaded.CreateEmptyState())
-		require.NoError(t, err)
-		path := writeToFile(t, "state.bin.gz", state)
-		version, err := DetectVersion(path)
-		require.NoError(t, err)
-		require.Equal(t, VersionSingleThreaded2, version)
-	})
+// Check that the latest supported versions write new states in a way that is detected correctly
+func TestDetectVersion_singleThreadedBinary(t *testing.T) {
+	targetVersion := VersionSingleThreaded2
+	if !arch.IsMips32 {
+		t.Skip("Single-threaded states are not supported for 64-bit VMs")
```

Comment:
nit: I think we shouldn't skip this. `DetectVersion` should reliably detect the appropriate version regardless of the cannon64 build tag. Though in practice, `DetectVersion` is only used by `multicannon` which isn't built for `cannon64`. Though this is more of a nit.